mysql-test/suite/innodb/t/innodb-alter-debug.test

```diff
@@ -77,3 +77,70 @@ drop table t1;
 
 # Wait till all disconnects are completed
 --source include/wait_until_count_sessions.inc
+
+--echo #
+--echo # MDEV-11415 AVOID INTERMEDIATE COMMIT WHILE DOING
+--echo #            ALTER TABLE...ALGORITHM=COPY
+--echo #
+
+CREATE DATABASE alter_db;
+USE alter_db;
+
+CREATE TABLE t(a SERIAL, b INT, c INT, d INT) ENGINE=InnoDB;
+INSERT INTO t(a) VALUES(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0);
+INSERT INTO t(a) SELECT a+16 FROM t;
+INSERT INTO t(a) SELECT a+33 FROM t;
+INSERT INTO t(a) VALUES(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0);
+INSERT INTO t(a) SELECT a+81 FROM t;
+INSERT INTO t(a) SELECT a+162 FROM t;
+INSERT INTO t(a) SELECT a+324 FROM t;
+INSERT INTO t(a) SELECT a+648 FROM t;
+INSERT INTO t(a) SELECT a+1296 FROM t;
+INSERT INTO t(a) SELECT a+2592 FROM t;
+INSERT INTO t(a) SELECT a+5186 FROM t;
+INSERT INTO t(a) VALUES(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0),(0);
+
+select count(*) from t;
+
+# try to make the to-be-created secondary index keys randomly distributed
+UPDATE t SET b=a%7, c=a%11, d=a%13;
+# crash right after the last write_row(), before the first commit of ALTER TABLE
+--source include/expect_crash.inc
+
+SET DEBUG_DBUG='+d,crash_copy_before_commit';
+--error 2013
+# create 32 secondary indexes
+ALTER TABLE t ADD INDEX(b,c,d,a),ADD INDEX(b,c,a,d),ADD INDEX(b,a,c,d),ADD INDEX(b,a,d,c),
+              ADD INDEX(b,d,a,c),ADD INDEX(b,d,c,a),ADD INDEX(a,b,c,d),ADD INDEX(a,b,d,c),
+              ADD INDEX(a,c,b,d),ADD INDEX(a,c,d,b),ADD INDEX(a,d,b,c),ADD INDEX(a,d,c,b),
+              ADD INDEX(c,a,b,d),ADD INDEX(c,a,d,b),ADD INDEX(c,b,a,d),ADD INDEX(c,b,d,a),
+              ADD INDEX(c,d,a,b),ADD INDEX(c,d,b,a),ADD INDEX(d,a,b,c),ADD INDEX(d,a,c,b),
+              ADD INDEX(d,b,a,c),ADD INDEX(d,b,c,a),ADD INDEX(d,c,a,b),ADD INDEX(d,c,b,a),
+              ADD INDEX(a,b,c), ADD INDEX(a,c,b), ADD INDEX(a,c,d), ADD INDEX(a,d,c),
+              ADD INDEX(a,b,d), ADD INDEX(a,d,b), ADD INDEX(b,c,d), ADD INDEX(b,d,c),
+              ALGORITHM=COPY;
+
+--source include/start_mysqld.inc
+
+SHOW CREATE TABLE t;
+CHECK TABLE t;
+DROP TABLE t;
+
+CREATE TABLE t1(a INT, b TEXT, c TEXT,
+                FULLTEXT(b), FULLTEXT(c(3)), FULLTEXT(b,c))ENGINE=InnoDB;
+INSERT INTO t1 VALUES(1, "This is a first b column", "This is a first c column");
```

Comment:
Minor nitpick: Standard SQL uses `'` for quoting string literals and `"` for quoting identifiers. MySQL and MariaDB use backticks for the latter, unless `ANSI_QUOTES` is set in `SQL_MODE`.
I would prefer to use standard syntax when possible, that is, use `'` instead of `"`.